pkg/clients/grpc/connect.go

```diff
@@ -49,14 +49,42 @@ var (
 	errFailedToLoadRootCA        = errors.New("failed to load root ca file")
 )
 
-type Config struct {
-	ClientCert        string        `env:"CLIENT_CERT"        envDefault:""`
-	ClientKey         string        `env:"CLIENT_KEY"         envDefault:""`
-	ServerCAFile      string        `env:"SERVER_CA_CERTS"    envDefault:""`
-	URL               string        `env:"URL"                envDefault:"localhost:7001"`
-	Timeout           time.Duration `env:"TIMEOUT"            envDefault:"60s"`
-	AttestedTLS       bool          `env:"ATTESTED_TLS"       envDefault:"false"`
-	AttestationPolicy string        `env:"ATTESTATION_POLICY" envDefault:""`
+type ClientConfiguration interface {
+	GetBaseConfig() BaseConfig
+}
+
+type BaseConfig struct {
+	URL          string        `env:"URL"             envDefault:"localhost:7001"`
+	Timeout      time.Duration `env:"TIMEOUT"         envDefault:"60s"`
+	ClientCert   string        `env:"CLIENT_CERT"     envDefault:""`
+	ClientKey    string        `env:"CLIENT_KEY"      envDefault:""`
+	ServerCAFile string        `env:"SERVER_CA_CERTS"    envDefault:""`
+}
+
+type ClientConfig struct {
+	BaseConfig
+}
+
+type AgentClientConfig struct {
+	ClientConfig
+	AttestationPolicy string `env:"ATTESTATION_POLICY" envDefault:""`
+	AttestedTLS       bool   `env:"ATTESTED_TLS"       envDefault:"false"`
+}
+
+type ManagerClientConfig struct {
+	ClientConfig
+}
+
+func (s ClientConfig) GetBaseConfig() BaseConfig {
+	return s.BaseConfig
+}
```

Comment:
why do you need both client and base config, clients are only two CLI(Agent) and manager